Client/core/Graphics/CMaterialPrimitive3DBatcher.h

```diff
@@ -0,0 +1,39 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto
+ *  LICENSE:     See LICENSE in the top level directory
+ *  FILE:        CMaterialPrimitive3DBatcher.h
+ *  PURPOSE:
+ *
+ *
+ *****************************************************************************/
+// Vertex type used by the primitives 3d batcher
+struct sPrimitiveMaterial3D
+{
+    D3DPRIMITIVETYPE                       eType;
+    CMaterialItem*                         pMaterial;
+    std::vector<PrimitiveMaterialVertice>* pVecVertices;
+};
+
+//
+// Batches primitives drawing
+//
+class CMaterialPrimitive3DBatcher
+{
+public:
+    ZERO_ON_NEW
+    CMaterialPrimitive3DBatcher(bool bPreGUI, CGraphics* graphics);
+    ~CMaterialPrimitive3DBatcher(void);
+    void OnDeviceCreate(IDirect3DDevice9* pDevice, float fViewportSizeX, float fViewportSizeY);
+    void Flush(void);
+    void AddPrimitive(D3DPRIMITIVETYPE eType, CMaterialItem* pMaterial, std::vector<PrimitiveMaterialVertice>* pVecVertices);
+    void DrawPrimitive(D3DPRIMITIVETYPE eType, size_t iCollectionSize, const void* pDataAddr, size_t uiVertexStride);
+    bool HasItems(void) { return !m_primitiveList.empty(); }
```

Comment:
Replace `(void)` with `()`. Make this member function **const**: `bool HasItems() const { ... }`